src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -330,15 +342,39 @@ def run_query(
         # because when the query plan has multiple queries, it will
         # have non-select statements, and it shouldn't fail if the user
         # calls to_pandas() to execute the query.
+        if not block:
+            return AsyncJob(
+                results_cursor["queryId"],
+                query,
+                self._conn,
```

Comment:
If you pass in the server connection object, you don't need to pass in self._conn.